File: apps/mobile/modules/walking-dog-watch-bridge/android/src/main/java/expo/modules/walkingdogwatchbridge/WalkingDogWatchBridgeModule.kt
package expo.modules.walkingdogwatchbridge

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class WalkingDogWatchBridgeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("WalkingDogWatchBridge")

    Events("onWatchWalkCommand")

    AsyncFunction("publishWalkSnapshot") { _: String ->
      Unit
    }

    AsyncFunction("getPendingCommands") {
      emptyList<String>()
    }

    AsyncFunction("ackCommand") { _: String ->
      Unit
    }
  }
}

File: apps/mobile/modules/walking-dog-watch-bridge/src/WalkingDogWatchBridgeModule.ts
import { NativeModule, requireNativeModule } from 'expo';

type WatchBridgeEvents = {
  onWatchWalkCommand(event: { commandJson: string }): void;
};

export type WatchWalkSnapshotPublishResult = {
  storedInAppGroup: boolean;
  watchConnectivitySupported: boolean;
  paired: boolean;
  watchAppInstalled: boolean;
  activationState: 'unsupported' | 'notActivated' | 'inactive' | 'activated' | 'unknown';
  reachable: boolean;
  activationRequested: boolean;
  applicationContextUpdated: boolean;
  immediateMessageAttempted: boolean;
  failureReason?: string;
  errorDescription?: string;
};

declare class WalkingDogWatchBridgeModule extends NativeModule<WatchBridgeEvents> {
  publishWalkSnapshot(snapshotJson: string): Promise<WatchWalkSnapshotPublishResult>;
  getPendingCommands(): Promise<string[]>;
  ackCommand(commandId: string): Promise<void>;
}

export default requireNativeModule<WalkingDogWatchBridgeModule>('WalkingDogWatchBridge');
